Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/impl/UserHelper.java

```diff
@@ -0,0 +1,52 @@
+package edu.asu.spring.quadriga.web.config.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.social.connect.Connection;
+import org.springframework.social.connect.UserProfile;
+import org.springframework.stereotype.Service;
+
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.domain.impl.User;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.IUserManager;
+import edu.asu.spring.quadriga.web.LoginController;
+import edu.asu.spring.quadriga.web.config.IUserHelper;
+
+@Service
+public class UserHelper implements IUserHelper{
+    
+    @Autowired
+    private IUserManager userManager;
+    private final Logger logger = LoggerFactory.getLogger(UserHelper.class);
+    @Override
+    public User createUser(Connection<?> connection) {
+        
+        UserProfile profile = connection.fetchUserProfile();
+        
+        String username = profile.getUsername() + "_" + connection.getKey().getProviderId();
+        
+        User user = new User();
+        
+        // make sure someone else didn't change their username to this one
+        IUser userWithUsername = null;
+        try {
+            userWithUsername = userManager.getUser(username);
+        } catch (QuadrigaStorageException e) {
+            logger.error("Error while fetching user details", e);
+        }
```

Comment:
if there was an exception thrown here, you will get a follow up nullpointerexception in line 39.